lib/git/git_ahead.fish

```diff
@@ -1,15 +1,23 @@
 function git_ahead -a ahead behind diverged none
-  git_is_repo; and begin
-    test -z "$ahead"; and set ahead "+"
-    test -z "$behind"; and set behind "-"
-    test -z "$diverged"; and set diverged "±"
-    test -z "$none"; and set none ""
-    command git rev-list --left-right "@{upstream}...HEAD" ^/dev/null \
-    | awk "/>/ {a += 1} /</ {b += 1} \
-      {if (a > 0) nextfile} END \
-      {if (a > 0 && b > 0) print \"$diverged\"; \
-      else if (a > 0) print \"$ahead\"; \
-      else if (b > 0) print \"$behind\";
-      else printf \"$none\"}"
+  not git_is_repo; and return
+
+  test -z "$ahead"; and set ahead "+"
+  test -z "$behind"; and set behind "-"
+  test -z "$diverged"; and set diverged "±"
+  test -z "$none"; and set none ""
+
+  set -l commit_count (command git rev-list --count --left-right "@{upstream}...HEAD" ^/dev/null)
+
+  switch "$commit_count"
+  case ""
+      # no upstream
+  case "0"\t"0"
+      echo "$none"
```

Comment:
I wonder if it would be faster to do test; and echo instead of doing it previously and assign it to a variable.